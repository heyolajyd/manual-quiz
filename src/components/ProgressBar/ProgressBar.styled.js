import styled from 'styled-components';

export const ProgressContainer = styled.div`
  height: 10px;
`;

export const Progress = styled.div`
  background-color: ${(props) => props.theme.colors.tealDark};
  height: inherit;
  width: ${(props) => props.width}%;
`;
