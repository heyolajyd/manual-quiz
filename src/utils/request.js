/**
 * Util function to get questions
 */
export const getQuestion = async () => {
  try {
    const res = await fetch('./questions.json');
    return await res.json();
  } catch (err) {
    console.log('Failed to parse: ', JSON.stringify(err));
    throw new Error(err);
  }
};
