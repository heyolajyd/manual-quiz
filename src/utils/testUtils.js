import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
