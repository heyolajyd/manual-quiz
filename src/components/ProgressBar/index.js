import React from 'react';
import PropTypes from 'prop-types';
import { ProgressContainer, Progress } from './ProgressBar.styled';

const ProgressBar = ({ questionNumber, questionCount }) => {
  const width =
    questionNumber && questionCount
      ? (questionNumber / questionCount) * 100
      : 0;

  return (
    <ProgressContainer>
      <Progress width={width} />
    </ProgressContainer>
  );
};

ProgressBar.propTypes = {
  questionNumber: PropTypes.number,
  questionCount: PropTypes.number,
};

export default ProgressBar;
