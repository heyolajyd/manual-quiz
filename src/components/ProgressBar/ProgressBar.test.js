import React from 'react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import ProgressBar from './index';

describe('ProgressBar', function () {
  beforeEach(() => {
    this.props = {
      questionNumber: 1,
      questionCount: 3,
    };
  });

  it('should match snapshot', () => {
    const { container } = withThemeProvider(<ProgressBar {...this.props} />);
    expect(container).toMatchSnapshot();
  });
});
