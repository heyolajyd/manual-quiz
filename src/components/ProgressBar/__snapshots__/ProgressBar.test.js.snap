// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshot 1`] = `
.c0 {
  height: 10px;
}

.c1 {
  background-color: #0c3c3d;
  height: inherit;
  width: 33.33333333333333%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="33.33333333333333"
    />
  </div>
</div>
`;
